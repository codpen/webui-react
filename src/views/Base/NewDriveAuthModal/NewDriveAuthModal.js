import React from "react";
import {Button, Modal, ModalBody, ModalFooter, ModalHeader} from "reactstrap";
import * as  PropTypes from "prop-types";


class NewDriveAuthModal extends React.Component {

    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
    }

    toggle() {
        this.props.closeModal();
    }


    render() {
        const {isVisible} = this.props;
        return (
            <div>
                <Modal isOpen={isVisible} toggle={this.toggle}>
                    <ModalHeader toggle={this.toggle} data-test="modalHeader">Configuring your drive.</ModalHeader>
                    <ModalBody data-test="modalBody">
                        A page will open for you with the authentication for your drive. This modal will automatically
                        dismiss upon successful creation
                    </ModalBody>
                    <ModalFooter data-test="modalFooter">
                        <Button color="primary" onClick={this.toggle}>Done</Button>{' '}
                        {/*<Button color="secondary" onClick={this.toggle}>Cancel</Button>*/}
                    </ModalFooter>
                </Modal>
            </div>
        );
    }
}

NewDriveAuthModal.propTypes = {
    isVisible: PropTypes.bool.isRequired,
    closeModal: PropTypes.func.isRequired
};
NewDriveAuthModal.defaultProps = {
    isVisible: true,
};


export default NewDriveAuthModal;