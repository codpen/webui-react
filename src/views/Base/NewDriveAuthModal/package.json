{
  "name": "NewDriveAuthModal",
  "version": "0.0.0",
  "private": true,
  "main": "NewDriveAuthModal.js"
}
