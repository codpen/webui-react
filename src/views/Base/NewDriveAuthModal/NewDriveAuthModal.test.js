import React from "react";
import NewDriveAuthModal from "./NewDriveAuthModal";
import {shallow} from "enzyme";
import {testStore} from "../../../../Utils";
import toJson from "enzyme-to-json";

const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);
    const component = shallow(<NewDriveAuthModal {...props} store={store}/>);
    return component.childAt(0).dive();
}

describe('New Drive Auth Modal', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const props = {
                isVisible: true,
                closeModal: jest.fn()
            };
            const initialState = {};
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            expect(wrapper).toHaveLength(1)
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });
    });


});