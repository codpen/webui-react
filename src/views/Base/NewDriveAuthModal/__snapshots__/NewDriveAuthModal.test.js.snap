// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Drive Auth Modal renders should match snapshot 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  container="body"
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  toggle={[Function]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    data-test="modalHeader"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Configuring your drive.
  </ModalHeader>
  <ModalBody
    data-test="modalBody"
    tag="div"
  >
    A page will open for you with the authentication for your drive. This modal will automatically dismiss upon successful creation
  </ModalBody>
  <ModalFooter
    data-test="modalFooter"
    tag="div"
  >
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Done
    </Button>
     
  </ModalFooter>
</Modal>
`;
